test/steps/application_v2_steps.py

```diff
@@ -621,28 +648,46 @@ def int_if_given(given):
     app_args = process_abi_args(
         context, context.abi_method, context.method_args
     )
+    context.app_args = app_args
     note = None
     if force_unique_transactions:
         note = (
             b"I should be unique thanks to this nonce: "
             + context.nonce.encode()
         )
 
-    context.atomic_transaction_composer.add_method_call(
-        app_id=app_id,
-        method=context.abi_method,
-        sender=sender,
-        sp=context.suggested_params,
-        signer=context.transaction_signer,
-        method_args=app_args,
-        on_complete=operation_string_to_enum(operation),
-        local_schema=local_schema,
-        global_schema=global_schema,
-        approval_program=approval_program,
-        clear_program=clear_program,
-        extra_pages=extra_pages,
-        note=note,
-    )
+    context.most_recent_added_method_exception = None
+    try:
+        context.atomic_transaction_composer.add_method_call(
+            app_id=app_id,
+            method=context.abi_method,
+            sender=sender,
+            sp=context.suggested_params,
+            signer=context.transaction_signer,
+            method_args=app_args,
+            on_complete=operation_string_to_enum(operation),
+            local_schema=local_schema,
+            global_schema=global_schema,
+            approval_program=approval_program,
+            clear_program=clear_program,
+            extra_pages=extra_pages,
+            note=note,
+        )
+    except AtomicTransactionComposerError as atce:
+        context.most_recent_added_method_exception = atce
+
+
+@then(
+    'the most recently added method call has an exception which satisfies "{regex}".'
```

Comment:
As a design note, generally we want to add error checking in the step that causes the error, like the step `I sign and submit the transaction, saving the txid. If there is an error it is "{error_string:MaybeString}".`

The reason being if there is a separate step to check for an error, it's all too easy to forget to check this step, meaning you will be silently ignoring potential errors.

Reply:
I think it makes sense to add more `I add a method call...` variants with a new `If there is an error it is...` suffix. That way existing `I add a method call...` usage does not need to change, there is just now a new version that should be used when you want to check for a specific error.

I believe all SDKs already use a common helper function to implement the 4 versions of this step, so adding potentially 4 more doesn't seem like a huge ask.

Is it weird to have 4+ variants of this step? Yes, but I think that's just the compromise we have to make when using cucumber.